Mathlib/RingTheory/PowerSeries/Restricted.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2025 William Coram. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: William Coram
+-/
+
+import Mathlib.Analysis.Normed.Group.Ultra
+import Mathlib.Analysis.RCLike.Basic
+import Mathlib.RingTheory.PowerSeries.Basic
+import Mathlib.Tactic.Bound
+
+/-!
+# Restricted power series
+
+`IsRestricted` : We say a powerseries over a normed ring `R` is restricted for a parameter `c` if
+`‖coeff R i f‖ * c ^ i → 0`.
+
+-/
+
+namespace PowerSeries
+
+open PowerSeries Filter
+open scoped Topology
+
+/-- A power series over `R` is restricted of paramerter `c` if we have
+  `‖coeff R i f‖ * c ^ i → 0`. -/
+def IsRestricted {R : Type*} [NormedRing R] (c : ℝ) (f : PowerSeries R) :=
+  Tendsto (fun (i : ℕ) => (norm (coeff R i f)) * c ^ i) atTop (𝓝 0)
+
+namespace IsRestricted
+
+variable {R : Type*} [NormedRing R] (c : ℝ)
+
+lemma isRestricted_iff {f : PowerSeries R} : IsRestricted c f ↔
+    ∀ ε, 0 < ε → ∃ N, ∀ n, N ≤ n → ‖‖(coeff R n) f‖ * c ^ n‖ < ε := by
+  simp [IsRestricted, NormedAddCommGroup.tendsto_atTop]
+
+lemma isRestricted_iff_abs (f : PowerSeries R) : IsRestricted c f ↔ IsRestricted |c| f := by
+  simp [isRestricted_iff]
+
+lemma zero : IsRestricted c (0 : PowerSeries R) := by
+  simp [IsRestricted]
+
+lemma one : IsRestricted c (1 : PowerSeries R) := by
+  simp only [isRestricted_iff, coeff_one, norm_mul, norm_pow, Real.norm_eq_abs]
+  refine fun _ _ ↦ ⟨1, fun n hn ↦ ?_ ⟩
+  split
+  · omega
+  · simpa
+
+lemma add {f g : PowerSeries R} (hf : IsRestricted c f) (hg : IsRestricted c g) :
```

Comment:
Can you add a result saying that any constant power series is restricted? Also we probably want the same for monomials.